sdk/src/main/java/ly/count/android/sdk/UtilsSdkInternalLimits.java

```diff
@@ -0,0 +1,97 @@
+package ly.count.android.sdk;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class UtilsSdkInternalLimits {
+
+    private UtilsSdkInternalLimits() {
+    }
+
+    /**
+     * This function is intended to be used to truncate the length of a key to a certain limit.
+     * It is used to ensure that the key length does not exceed the limit set by the SDK.
+     * If the key length exceeds the limit, the key is truncated to the limit.
+     * If the key length is less than or equal to the limit, the key is returned as is.
+     * Used truncate method is substring. from 0 to limit.
+     * <pre>
+     * Intended to be used for those:
+     * - event names
+     * - view names
+     * - custom trace key name (APM)
+     * - custom metric key (APM)
+     * - segmentation key (for all features)
+     * - custom user property
+     * - custom user property keys that are used for property modifiers (mul, push, pull, set, increment, etc)
+     * </pre>
+     *
+     * @param key to truncate
+     * @param limit to truncate to
+     * @param L logger
+     * @return truncated key
+     */
+    protected static String truncateKeyLength(@Nullable String key, final int limit, @NonNull ModuleLog L) {
```

Comment:
a stirng tag/message is needed to indicate the origin of this call. It should be used in all logs related to truncating